barbell/src/barbell/bot.ts

```diff
@@ -327,20 +316,20 @@ class ActionRunner {
 
 export class Action {
 	readonly name: string
-	readonly handler: (io: IO) => Promise<void>
+	readonly handler: (handlerInputs: HandlerInput) => Promise<void>
 
-	constructor({ name, handler }: { name: string, handler: (io: IO) => Promise<void> }) {
+	constructor({ name, handler }: { name: string, handler: (handlerInputs: HandlerInput) => Promise<void> }) {
 		this.name = name
 		this.handler = handler
 	}
-	async run(inputs?: Record<string, { type: string, [key: string]: any }> | undefined, buttonClick?: { action: string, value: string } | undefined) {
+	async run(userId: string, channelId: string, channelType: ChannelType, inputs?: Record<string, { type: string, [key: string]: any }> | undefined, buttonClick?: { action: string, value: string } | undefined) {
```

Comment:
we may want to encapsulate these values. This is optional unless we reference a lot more

Reply:
We can just pass HandlerInput

Reply:
i agree this is verbose, we can always refactor later haha